ui/docs-app/docs_app/theme/assets/semantic-ui/js/docs_app/forms/components/FileUploader/FileUploaderWrappers.jsx

```diff
@@ -0,0 +1,94 @@
+import React, { useEffect, useRef } from "react";
+import { h, render } from "preact";
+import PropTypes from "prop-types";
+import { i18next } from "@translations/docs_app/i18next";
+
+export const FileUploadWrapper = ({
+  preactComponent,
+  uploadWrapperClassName,
+  uploadButtonClassName,
+  props,
+}) => {
+  const preactCompRef = useRef();
+  useEffect(() => {
+    render(
```

Comment:
This Uppy usage looks really wild, hard to maintain & understand. I think we should actually be using existing React components & hooks https://uppy.io/docs/react/ to implement the uppy uploader dialog, are there any blockers preventing us from using it (@corovcam)? From docs, it looks like any of the `preact` magic shouldn't be needed here :)

Reply:
There's one big issue. When building custom Uppy plugins, rendered inside Uppy Dashboard, Uppy uses Preact components and hooks. Meaning, Preact is bundled inside the package and there's no alternative (preact/compat doesn't work). The Preact hooks are incompatible with React, so if React calls Preact hook (in custom plugin) in repository, error is issued.